lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -107,11 +107,12 @@ sub query {
         IPC::Run::run(\@cmd, \undef, \$stdout, \$stderr);
         return $self->render(json => {error => "Grep failed: $stderr"}, status => 400) if $stderr;
 
-        my @lines = split("\n", $stdout);
+        my $basename = $distri->basename;
+        my @lines    = split("\n", $stdout);
         splice @lines, $cap;
         foreach my $match (@lines) {
             my ($filename, $occurrence) = split(':', $match);
-            push(@results, {occurrence => $distri->basename . '/' . $filename, contents => $occurrence});
+            push(@results, {occurrence => "$basename/$filename", contents => $occurrence}) if $filename;
```

Comment:
I don't understand the purpose of this. Each line is `filename:contents`. Unless perhaps you're seeing a side effect of `:` being parsed incorrectly if the line contains `:`, which I'm addressing in #3317 by adding a limit to the `split` call.

Reply:
If there are no search results a single line break might be returned. In this case  `$match` is empty. In this case `$filename` is not defined leading to `Use of uninitialized $filename` as stated in the commit message.

(And yes, `split` apparently not just returns an empty string here, e.g.
```
  DB<5> ($filename, $foo) = split(':', '');
  DB<6> print('not defined') unless defined $filename;
not defined
```
)

Reply:
Then it might be even better to do:
```
next unless length $match;
```
before the split.

Reply:
Ah. And that's not even a proper failure, which explains why the tests didn't catch it. 

Although in that case I'd rather do a `strip` or something else to avoid the loop. Semantically no filename just makes no sense.

Reply:
I guess I follow @perlpunk's suggestion. That seems to be the simplest way to achieve the correct behavior.